crates/swc_css_modules/tests/with-compat/vercel/001/output.css

```diff
@@ -7,7 +7,7 @@
   border: 1px solid var(--accents-2);
 }
 .dark-theme .__local__input {}
-.dark-theme .__local__input::placeholder {
+.dark-theme .__local__input:placeholder {
```

Comment:
@alexander-akait I'm not sure if this is correct

Reply:
Yeah, it is invalid, `placeholder` is pseudo element, not pseudo class

Can you descibe the original problem?

Reply:
The problem is 

```css
.gradient-button-wrapper {
    /*  animated border gradient */
    .button-bg {
        &,
        &::before {
            background-image: linear-gradient(
                165deg,
                var(--start-color),
                var(--end-color)
            );
        }
    }
}
```

is compiled differently from postcss plugin.

```css
.gradient-button-wrapper .button-bg:before
```

is the expected output

Reply:
Maybe I misunderstood the issue.

Reply:
Yeah, current;y all works fine, maybe an issue about transformation `::before` -> `:before` for old IE

Reply:
@alexander-akait Can you check [playground](https://lightningcss.dev/playground/index.html#%7B%22minify%22%3Afalse%2C%22nesting%22%3Atrue%2C%22customMedia%22%3Atrue%2C%22cssModules%22%3Afalse%2C%22analyzeDependencies%22%3Afalse%2C%22targets%22%3A%7B%22chrome%22%3A10223616%7D%2C%22source%22%3A%22.gradient-button-wrapper%20%7B%5Cn%20%20.gradient-button%20%7B%5Cn%20%20%20%20transition-timing-function%3A%20ease%3B%5Cn%5Cn%20%20%20%20%3Aglobal(.dark-theme)%20%26%20%7B%5Cn%20%20%20%20%20%20box-shadow%3A%200px%204px%204px%200px%20%2300000040%3B%5Cn%20%20%20%20%7D%5Cn%5Cn%20%20%20%20%26%5Bdata-hover%5D%20%7B%5Cn%20%20%20%20%20%20--lighten-color%3A%20transparent%3B%5Cn%5Cn%20%20%20%20%20%20color%3A%20var(--geist-background)%3B%5Cn%20%20%20%20%20%20background-color%3A%20transparent%3B%5Cn%20%20%20%20%7D%5Cn%5Cn%20%20%20%20%26%5Bdata-active%5D%20%7B%5Cn%20%20%20%20%20%20--lighten-color%3A%20rgba(255%2C%20255%2C%20255%2C%200.5)%3B%5Cn%20%20%20%20%7D%5Cn%5Cn%20%20%20%20%26%5Bdata-focus%5D%20%7B%5Cn%20%20%20%20%20%20box-shadow%3A%200%200%200%201px%20var(--geist-background)%2C%5Cn%20%20%20%20%20%20%20%200%200%200%203px%20var(--themed-bg%2C%20var(--geist-foreground))%3B%5Cn%20%20%20%20%7D%5Cn%20%20%7D%5Cn%5Cn%20%20%2F*%20%20animated%20border%20gradient%20*%2F%5Cn%20%20.button-bg%20%7B%5Cn%20%20%20%20border-radius%3A%20var(--geist-radius)%3B%5Cn%20%20%20%20content%3A%20''%3B%5Cn%20%20%20%20height%3A%20100%25%3B%5Cn%20%20%20%20position%3A%20absolute%3B%5Cn%20%20%20%20width%3A%20100%25%3B%5Cn%20%20%20%20z-index%3A%20-2%3B%5Cn%5Cn%20%20%20%20%2F*%20animated%20glow%20*%2F%5Cn%20%20%20%20%26%3A%3Abefore%20%7B%5Cn%20%20%20%20%20%20border%3A%2012px%20solid%20transparent%3B%5Cn%20%20%20%20%20%20background-clip%3A%20padding-box%3B%5Cn%20%20%20%20%20%20content%3A%20''%3B%5Cn%20%20%20%20%20%20filter%3A%20blur(36px)%3B%5Cn%20%20%20%20%20%20height%3A%20100%25%3B%5Cn%20%20%20%20%20%20opacity%3A%200.8%3B%5Cn%20%20%20%20%20%20position%3A%20absolute%3B%5Cn%20%20%20%20%20%20width%3A%20100%25%3B%5Cn%20%20%20%20%20%20z-index%3A%20-1%3B%5Cn%5Cn%20%20%20%20%20%20%3Aglobal(.dark-theme)%20%26%20%7B%5Cn%20%20%20%20%20%20%20%20opacity%3A%201%3B%5Cn%20%20%20%20%20%20%7D%5Cn%20%20%20%20%7D%5Cn%5Cn%20%20%20%20%26%2C%5Cn%20%20%20%20%26%3A%3Abefore%20%7B%5Cn%20%20%20%20%20%20background-image%3A%20linear-gradient(%5Cn%20%20%20%20%20%20%20%20165deg%2C%5Cn%20%20%20%20%20%20%20%20var(--start-color)%2C%5Cn%20%20%20%20%20%20%20%20var(--end-color)%5Cn%20%20%20%20%20%20)%3B%5Cn%20%20%20%20%7D%5Cn%5Cn%20%20%7D%5Cn%7D%5Cn%22%2C%22visitorEnabled%22%3Afalse%2C%22visitor%22%3A%22%7B%5Cn%20%20Color(color)%20%7B%5Cn%20%20%20%20if%20(color.type%20%3D%3D%3D%20'rgb')%20%7B%5Cn%20%20%20%20%20%20color.g%20%3D%200%3B%5Cn%20%20%20%20%20%20return%20color%3B%5Cn%20%20%20%20%7D%5Cn%20%20%7D%5Cn%7D%22%2C%22unusedSymbols%22%3A%5B%5D%2C%22version%22%3A%22local%22%7D)?

Seems like it's compiling it as `:before` even for latest chrome

Reply:
Because it's shorter, `:before` with one `:` is only for IE8 and IE9 (and less, other browsers support it for compatibility reasons), i.e. in your code you should not transform pseudo element to pleudo classes, it is wrong, our minifier already compress `::before` -> `:before`, so you don't need to worry about size.

If we will need support IE8 and IE9 we will implement tranformation from `::before` -> `:before` (and only for `::before/::after/::first-letter/::first-line`), but this issue is not about it, this issue about unnecessary (wring) handling pseudo classes and pseudo elements in nesting